### PR TITLE
#848 Add faq categories

### PR DESCRIPTION
**Pull request by alexong7**
# Description
Add FAQ categories. Categories are set in an FAQ array and added to the FAQ items separated by the regex #####.

(Note this is for the purposes of a school assignment to add to an open source git issue, not entirely perfect, but provides a starting solution to be iterated on)

Screen Shot 2023-04-25 at 7 58 27 PM

Screen Shot 2023-04-25 at 7 58 30 PM

Fixes #(issue)
#848 

## Type of change
Just put an x in the [] which are valid.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ x ] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.
- [ ] `./gradlew assembleDebug assembleRelease`
- [ ] `./gradlew checkstyle`

# Checklist:
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings

### CODE REVIEW

File: app/src/main/java/swati4star/createpdf/fragment/FAQFragment.java [hunk 1]
```diff
@@ -42,6 +47,41 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
...
+        String[] categories = mContext.getResources().getStringArray(R.array.faq_categories);
+        TabLayout tabLayout = (TabLayout) view.findViewById(R.id.faq_category_tab_layout);
+
+        // For FAQ items without a category
+        tabLayout.addTab(tabLayout.newTab().setText("All"));
+
+        for (String category : categories) tabLayout.addTab(tabLayout.newTab().setText(category));
+
+
+        tabLayout.setOnTabSelectedListener(new TabLayout.OnTabSelectedListener() {
```
Comment (alexong7) [inline app/src/main/java/swati4star/createpdf/fragment/FAQFragment.java:64]: This is deprecated. With more time I would implement an updated version and clean up some of this code

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -580,29 +580,34 @@
     <string name="storage_manager_permission_rationale">In Android 11, you need to allow a special storage permission</string>
     <string name="close_app_text">Close App</string>
     <string name="allow_text">Allow</string>
+    <string-array name="faq_categories">
```
Comment (alexong7) [inline app/src/main/res/values/strings.xml:583]: Placeholder Categories. Once again, this isn't a final product, just for purposes of a school assignment but a good starting point

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -580,29 +580,34 @@
...
+        <item>Category 1</item>
+        <item>Category 2</item>
+        <item>Category 3</item>
+    </string-array>
     <string-array name="faq_question_answers">
-        <item>What image file formats are supported?#####The app supports jpeg, jpg, tiff, gif, psd, bmp, eps, png, etc.</item>
+        <item>What image file formats are supported?#####The app supports jpeg, jpg, tiff, gif, psd, bmp, eps, png, etc.#####Category 1</item>
```
Comment (alexong7) [inline app/src/main/res/values/strings.xml:589]: Once again, placeholder Categories. But this provides the general ideas

File: app/src/main/res/values/strings.xml (hunk 2 above)
Comment (Swati4star) [reply app/src/main/res/values/strings.xml:583]: The current repository is a final product and we want actual informative chnages here. We can't add placeholder changes here currently.

**Pull request closed without merging**